src/main/java/com/github/lombrozo/jsmith/antlr/rules/LabeledElement.java

```diff
@@ -33,19 +40,63 @@
  * }
  * @since 0.1
  */
-public final class LabeledElement extends Unimplemented {
+public final class LabeledElement implements Rule {
 
     /**
      * Labeled element name.
      */
     private static final String ALIAS = "labeledElement";
 
+    /**
+     * Parent rule.
+     */
+    private final Rule top;
+
+    /**
+     * List of elements.
+     */
+    private final List<Rule> elements;
+
     /**
      * Constructor.
      * @param parent Parent rule.
      */
     public LabeledElement(final Rule parent) {
-        super(parent);
+        this(parent, new ArrayList<>(0));
+    }
+
+    /**
+     * Constructor.
+     * @param parent Parent rule
+     * @param elements List of elements
+     */
+    public LabeledElement(final Rule parent, final List<Rule> elements) {
+        this.top = parent;
+        this.elements = elements;
+    }
+
+    @Override
+    public Rule parent() {
+        return this.top;
+    }
+
+    @Override
+    public Node generate(final Context context) throws WrongPathException {
+        if (this.elements.size() != 3) {
```

Comment:
@RechnoyBobr I saw the similar code in other classes. Maybe we can do the same for it as we did for the `Allowed` class - extract to a separate class and reuse?